lecture-notes/3-bld.md

```diff
@@ -680,31 +680,271 @@ $
 
 you can use `HELLO` as a Make variable.
 
-In short, environment variables (e.g. `export CFLAGS=-g ; make hello`):
+In short, when using environment variables in commands such as `export CFLAGS=-g ; make hello`:
 
-* Tells the shell to run make with `CFLAGS` set in the environment
+* The shell runs make with `CFLAGS` set in the environment
 * Make imports all environment variables as Make variables
 * However, assignments in the Makefile override environment variables! `=` and
   `:=` replace them entirely, while `+=` appends to them
 
+So, for example, the following Makefile overrides the environment variable:
+
+```make
+.PHONY: all
+HELLO:=Max
+all:
+	echo $(HELLO)
+```
+
+```console
+$ HELLO=Tom make
+echo Max
+Max
+$
+```
+
 #### Variable overrides
 
-<!-- TODO -->
+Variable overrides are different. While environment variables allow for
+customization from the outside, variable overrides are a feature of Make that
+let the user forcibly change variable values.
+
+This means that making changes to variables need not involve mucking around
+inside a Makefile and finding all places where a variable is initialized.
+Instead, you can decide once and for all what the value should be when invoking
+`make`.
+
+Imagine using someone else's Makefile where they have hard-coded `CC:=gcc` at
+the top. No need to edit; run `make CC=clang` and enjoy using your compiler of
+choice.
 
-In short, variable overrides (e.g. `make CFLAGS=-g hello`):
+In short, when using variable overrides in commands such as `make CFLAGS=-g hello`:
 
-* *Not* an environment variable: value passed as an argument to `make` directly
-* Sets the given Make variable to the given value
+* The shell does *not* use an environment variable: the shell passes the string
+  `CFLAGS=-g` as an argument to `make`, which parses it
+* Make sets the given variable to the given value
 * Causes *all* assignments in Makefile to that variable (`=`, `:=`, and `+=`)
   to be ignored
 
-### "multi-target" rules
+So, for example, the following Makefile's `HELLO` value is ignored in favor of
+the override:
+
+```make
+.PHONY: all
+HELLO:=Max
+all:
+	echo $(HELLO)
+```
+
+```console
+$ make HELLO=Tom
+echo Max
+Max
+$
+```
+
 ### Implicit variables
+
+Some variables used by implicit rules, like `CC` (the C compiler) have default
+values (`CC`, for example, defaults to `"cc"`). While they exist for use
+primarily in implicit rules, they also have values all throughout the Makefile;
+use them as you see fit.
+
+They are different than "normal" variables in two ways:
+
+* These default values have lower precedence than environment variables
+* Can be overridden by the environment or an assignment in your Makefile
+
+So, for example, the following Makefile's (default, implicit) `CC` value of
+`"cc"` is ignored in favor of the environment variable:
```

Comment:
Maybe an example of just running `make` to show that it outputs `cc` by default? Unless you think that's too verbose.